File: Examples/WindowsFormsProject/Form1.cs
using System;
using System.Linq;
using System.Windows;
using System.Windows.Forms;
using GraphX.PCL.Common.Enums;
using GraphX.PCL.Logic.Algorithms.OverlapRemoval;
using GraphX.PCL.Logic.Models;
using GraphX.Controls;
using QuickGraph;

namespace WindowsFormsProject
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            Load += Form1_Load;
        }

        void Form1_Load(object sender, EventArgs e)
        {
            wpfHost.Child = GenerateWpfVisuals();
            _zoomctrl.ZoomToFill();
        }

        private ZoomControl _zoomctrl;
        private GraphAreaExample _gArea;

        private UIElement GenerateWpfVisuals()
        {
            _zoomctrl = new ZoomControl();
            ZoomControl.SetViewFinderVisibility(_zoomctrl, Visibility.Visible);
            /* ENABLES WINFORMS HOSTING MODE --- >*/
            var logic = new GXLogicCore<DataVertex, DataEdge, BidirectionalGraph<DataVertex, DataEdge>>();
            _gArea = new GraphAreaExample() { EnableWinFormsHostingMode = true, LogicCore = logic };
            logic.Graph = GenerateGraph();
            logic.DefaultLayoutAlgorithm = LayoutAlgorithmTypeEnum.LinLog;
            logic.DefaultLayoutAlgorithmParams = logic.AlgorithmFactory.CreateLayoutParameters(LayoutAlgorithmTypeEnum.LinLog);
            //((LinLogLayoutParameters)logic.DefaultLayoutAlgorithmParams). = 100;
            logic.DefaultOverlapRemovalAlgorithm = OverlapRemovalAlgorithmTypeEnum.FSA;
            logic.DefaultOverlapRemovalAlgorithmParams = logic.AlgorithmFactory.CreateOverlapRemovalParameters(OverlapRemovalAlgorithmTypeEnum.FSA);
            ((OverlapRemovalParameters)logic.DefaultOverlapRemovalAlgorithmParams).HorizontalGap = 50;
            ((OverlapRemovalParameters)logic.DefaultOverlapRemovalAlgorithmParams).VerticalGap = 50;
            logic.DefaultEdgeRoutingAlgorithm = EdgeRoutingAlgorithmTypeEnum.None;
            logic.AsyncAlgorithmCompute = false;
            _zoomctrl.Content = _gArea;
            _gArea.RelayoutFinished += gArea_RelayoutFinished;

            
            var myResourceDictionary = new ResourceDictionary {Source = new Uri("Templates\\template.xaml", UriKind.Relative)};
            _zoomctrl.Resources.MergedDictionaries.Add(myResourceDictionary);

            return _zoomctrl;
        }

        void gArea_RelayoutFinished(object sender, EventArgs e)
        {
            _zoomctrl.ZoomToFill();
        }

        private GraphExample GenerateGraph()
        {
            //FOR DETAILED EXPLANATION please see SimpleGraph example project
            var dataGraph = new GraphExample();
            for (int i = 1; i < 10; i++)
            {
                var dataVertex = new DataVertex("MyVertex " + i);
                dataGraph.AddVertex(dataVertex);
            }
            var vlist = dataGraph.Vertices.ToList();
            //Then create two edges optionaly defining Text property to show who are connected
            var dataEdge = new DataEdge(vlist[0], vlist[1]) { Text = string.Format("{0} -> {1}", vlist[0], vlist[1]) };
            dataGraph.AddEdge(dataEdge);
            dataEdge = new DataEdge(vlist[2], vlist[3]) { Text = string.Format("{0} -> {1}", vlist[2], vlist[3]) };
            dataGraph.AddEdge(dataEdge);


            dataEdge = new DataEdge(vlist[2], vlist[2]) { Text = string.Format("{0} -> {1}", vlist[2], vlist[2]) };
            dataGraph.AddEdge(dataEdge);
            return dataGraph;
        }

        private void but_generate_Click(object sender, EventArgs e)
        {
            _gArea.GenerateGraph(true);
            _gArea.SetVerticesDrag(true, true);
            _zoomctrl.ZoomToFill();
        }

        private void but_reload_Click(object sender, EventArgs e)
        {
            _gArea.RelayoutGraph();
        }
    }
}

File: Examples/WindowsDesktop_VB.NET_WinForms_Example/Form1.vb
Imports System.Windows
Imports WindowsDesktop_VB.NET_WinForms_Example.Models
Imports GraphX.PCL.Common.Enums
Imports GraphX.PCL.Logic.Algorithms.LayoutAlgorithms
Imports GraphX.PCL.Logic.Algorithms.OverlapRemoval
Imports GraphX.PCL.Logic.Models
Imports GraphX.Controls
Imports QuickGraph

Public Class Form1

    Dim _zoomctrl As ZoomControl
    Dim _gArea As GraphAreaExample

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        wpfHost.Child = GenerateWpfVisuals()

        _gArea.GenerateGraph(True)

        _zoomctrl.ZoomToFill()
    End Sub

    Private Function GenerateWpfVisuals() As UIElement
        _zoomctrl = New ZoomControl()
        ZoomControl.SetViewFinderVisibility(_zoomctrl, Visibility.Visible)
        ' ENABLES WINFORMS HOSTING MODE --- >

        Dim logic = New GXLogicCore(Of DataVertex, DataEdge, BidirectionalGraph(Of DataVertex, DataEdge))()
        _gArea = New GraphAreaExample()
        _gArea.EnableWinFormsHostingMode = True
        _gArea.LogicCore = logic

        logic.Graph = GenerateGraph()
        logic.DefaultLayoutAlgorithm = LayoutAlgorithmTypeEnum.KK
        logic.DefaultLayoutAlgorithmParams = logic.AlgorithmFactory.CreateLayoutParameters(LayoutAlgorithmTypeEnum.KK)
        DirectCast(logic.DefaultLayoutAlgorithmParams, KKLayoutParameters).MaxIterations = 100
        logic.DefaultOverlapRemovalAlgorithm = OverlapRemovalAlgorithmTypeEnum.FSA
        logic.DefaultOverlapRemovalAlgorithmParams = logic.AlgorithmFactory.CreateOverlapRemovalParameters(OverlapRemovalAlgorithmTypeEnum.FSA)
        DirectCast(logic.DefaultOverlapRemovalAlgorithmParams, OverlapRemovalParameters).HorizontalGap = 50
        DirectCast(logic.DefaultOverlapRemovalAlgorithmParams, OverlapRemovalParameters).VerticalGap = 50
        logic.DefaultEdgeRoutingAlgorithm = EdgeRoutingAlgorithmTypeEnum.None
        logic.AsyncAlgorithmCompute = False
        _zoomctrl.Content = _gArea
        Dim myResourceDictionary = New ResourceDictionary()
        myResourceDictionary.Source = New Uri("Templates\template.xaml", UriKind.Relative)
        _zoomctrl.Resources.MergedDictionaries.Add(myResourceDictionary)

        Return _zoomctrl
    End Function

    Private Function GenerateGraph() As BidirectionalGraph(Of DataVertex, DataEdge)
        'FOR DETAILED EXPLANATION please see SimpleGraph example project
        Dim dataGraph = New GraphExample()
        For i As Integer = 1 To 9
            Dim dataVertex = New DataVertex("MyVertex " & i)
            dataGraph.AddVertex(dataVertex)
        Next
        Dim vlist = dataGraph.Vertices.ToList()
        'Then create two edges optionaly defining Text property to show who are connected
        Dim dataEdge = New DataEdge(vlist(0), vlist(1))
        dataEdge.Text = String.Format("{0} -> {1}", vlist(0), vlist(1))
        dataGraph.AddEdge(dataEdge)
        dataEdge = New DataEdge(vlist(2), vlist(3))
        dataEdge.Text = String.Format("{0} -> {1}", vlist(2), vlist(3))
        dataGraph.AddEdge(dataEdge)

        dataEdge = New DataEdge(vlist(2), vlist(2))
        dataEdge.Text = String.Format("{0} -> {1}", vlist(2), vlist(2))
        dataGraph.AddEdge(dataEdge)

        Return dataGraph
    End Function
End Class
